src/http.rs

```diff
@@ -123,6 +124,23 @@ where
                 })
                 .context(CallRequestSnafu)?;
 
+            let (parts, body) = response.into_parts();
+            let mut data_buf: Option<Vec<u8>> = Some(Vec::new());
+
+            let body = body
+                .map_data(move |data| {
+                    if let Some(buf) = &mut data_buf {
+                        buf.extend_from_slice(&data);
+                        if buf.len() >= 20 {
+                            emit!(http_client::HttpBodyDataReceived { buf });
```

Comment:
The truncation can be moved here by using `buf: &buf[..20]`

Reply:
I opted to use the buf.truncate(20) method since there was no complaints from the compiler implementing the &buf[..20] 

If buf.truncate() is not acceptable I will make the change before this pr is merged